test/active_record_doctor/detectors/missing_unique_indexes_test.rb

```diff
@@ -179,12 +179,58 @@ def test_column_order_is_ignored
     refute_problems
   end
 
-  def test_conditions_is_skipped
-    assert_skipped(conditions: -> { where.not(email: nil) })
+  def test_case_insensitive_index_exists
+    skip("Expression indexes are not supported") if expression_indexes_unsupported?
+
+    create_table(:users) do |t|
+      t.string :email
+
+      t.index :email, unique: true
+    end.create_model do
+      validates :email, uniqueness: { case_sensitive: false }
+    end
+
+    assert_problems(<<~OUTPUT)
+      add a unique index on users(lower(email)) - validating uniqueness in the model without an index can lead to duplicates
+    OUTPUT
   end
 
-  def test_case_insensitive_is_skipped
-    assert_skipped(case_sensitive: false)
+  def test_case_insensitive_lower_index_exists
+    skip("Expression indexes are not supported") if expression_indexes_unsupported?
+
+    create_table(:users) do |t|
+      t.string :email
+    end.create_model do
+      validates :email, uniqueness: { case_sensitive: false }
+    end
+
+    # ActiveRecord < 5 does not support expression indexes.
+    ActiveRecord::Base.connection.execute(<<-SQL)
+      CREATE UNIQUE INDEX index_users_on_lower_email ON users ((lower(email)))
+    SQL
+
+    refute_problems
+  end
+
+  def test_case_insensitive_compound_lower_index_exists
+    skip("Expression indexes are not supported") if expression_indexes_unsupported?
+
+    create_table(:users) do |t|
+      t.string :email
+      t.integer :organization_id
```

Comment:
I think in this case we also should a unique index on `(organization_id, email)` to ensure only an **expression** index will satisfy the condition. Similarily, we need another test case with a non-unique expression index.